### ISSUE TITLE
 Arabic font enhancment

### ISSUE THREAD

**Issue by AhmadDakhlallah** [nextcloud-libraries/nextcloud-vue#3780]
Hello,

I've noticed that there is no Arabic font support. The system will use the default system font which is OK at best.
I'd like to suggest using some font like "Tajawal" which is the best at this time for Arabic:
https://fonts.google.com/specimen/Tajawal?query=tajawal

Other suggested fonts:

https://fonts.google.com/specimen/IBM+Plex+Sans+Arabic?subset=arabic&noto.script=Arab
https://fonts.google.com/specimen/Cairo?subset=arabic&noto.script=Arab
https://fonts.google.com/specimen/Vazirmatn?subset=arabic&noto.script=Arab

You can apply the font in 2 ways, the less preferred way is to follow the interface language and set fallback for other languages, or the better way to detect utf8 characters and apply font for each language independently.

Thank you!!

**Comment by jancborchardt** [comment]
Hi @AhmadDakhlallah, thank you for the feedback! We especially moved to a native font stack a while ago, so we don't ship any fonts.

Is there any (or multiple) widely preinstalled Arabic fonts we could just add to the font stack?

Also cc @hamza221 @mabkenar as they might know more. :)

**Comment by AhmadDakhlallah** [comment]
Hey,
Thanks for the reply!

While windows does implement a good font called SgoeUI or the older Tahome font, Linux does implement the Noto font which isn't good when used in web applications. The Noto font doesn't look or feel as polished for screen as the fonts I've mentioned and we can't try to convince all people to install external fonts to make the application looks nice on their devices.

For your reply, I understand that we can add support for a specific font present by default on the user's OS, but we can't add an additional webfont to do the same effect.

Is there a way to inject a custom css to do the same without actually editing core files?

**Comment by susnux** [comment]
> Is there a way to inject a custom css to do the same without actually editing core files?

Yes see this app https://apps.nextcloud.com/apps/theming_customcss

**Comment by susnux** [comment]
closing as this needs to be done on server
